docs/modules/ROOT/pages/access.adoc

```diff
@@ -1,672 +1,421 @@
+In the Solidity implementation of AccessControl, contracts generally refer to the
+https://docs.soliditylang.org/en/latest/units-and-global-variables.html?highlight=keccak256#mathematical-and-cryptographic-functions[keccak256 hash]
+of a role as the role identifier.
+
 For example:
 
-[,solidity]
+[,javascript]
 ----
 bytes32 public constant SOME_ROLE = keccak256("SOME_ROLE")
 ----
 
 These identifiers take up 32 bytes (256 bits).
 
-Cairo field elements store a maximum of 252 bits.
-Even further, a declared _constant_ field element in a StarkNet contract stores even less (see https://github.com/starkware-libs/cairo-lang/blob/release-v0.6.1/src/starkware/cairo/lang/cairo_constants.py#L1[cairo_constants]).
+Cairo field elements (`felt252`) store a maximum of 252 bits.
 With this discrepancy, this library maintains an agnostic stance on how contracts should create identifiers.
 Some ideas to consider:
 
-* Use the first or last 251 bits of keccak256 hash digests.
-* Use Cairo's https://github.com/starkware-libs/cairo-lang/blob/master/src/starkware/cairo/common/hash.cairo[hash2].
-
-=== AccessControl library API
-
-[,cairo]
-----
-func initializer() {
-}
-
-func assert_only_role(role: felt) {
-}
-
-func has_role(role: felt, user: felt) -> (has_role: felt) {
-}
-
-func get_role_admin(role: felt) -> (admin: felt) {
-}
-
-func grant_role(role: felt, user: felt) {
-}
-
-func revoke_role(role: felt, user: felt) {
-}
-
-func renounce_role(role: felt, user: felt) {
-}
-
-func _grant_role(role: felt, user: felt) {
-}
-
-func _revoke_role(role: felt, user: felt) {
-}
-
-func _set_role_admin(role: felt, admin_role: felt) {
-}
-----
-
-[#initializer-accesscontrol]
-==== `initializer`
-
-Initializes AccessControl and should be called in the implementing contract's constructor.
-
-This must only be called once.
-
-Parameters: None.
-
-Returns: None.
-
-==== `assert_only_role`
-
-Checks that an account has a specific role.
-Reverts with a message including the required role.
-
-Parameters:
-
-[,cairo]
-----
-role: felt
-----
-
-Returns: None.
-
-==== has_role
-
-Returns `TRUE` if `user` has been granted `role`, `FALSE` otherwise.
-
-Parameters:
-
-[,cairo]
-----
-role: felt
-user: felt
-----
-
-Returns:
-
-[,cairo]
-----
-has_role: felt
-----
-
-==== `get_role_admin`
+* Use the {sn_keccak} instead.
+* Use Cairo friendly hashing algorithms like Poseidon, which are implemented in the
+https://github.com/starkware-libs/cairo/blob/main/corelib/src/poseidon.cairo[Cairo corelib].
 
-Returns the admin role that controls `role`.
-See <<grant_role,grant_role>> and <<revoke_role,revoke_role>>.
+TIP: The `selector!` macro can be used for computing the {sn_keccak} in Cairo.
 
-To change a role's admin, use <<set_role_admin,`_set_role_admin`>>.
+=== Interface
 
-Parameters:
+This is the full interface a contract must implement to be compliant with the `AccessControl` implementation:
```

Comment:
what does it mean to be compliant with the impl? isn't with the standard? although we don't mention any standard for this.

i think it's especially confusing to have a single function described below (`renounce_role`).

Reply:
> what does it mean to be compliant with the impl? isn't with the standard?

With our implementation of AccessControl, I think using the word standard without a standard will be confusing.

> i think it's especially confusing to have a single function described below (renounce_role).

The guide already explained the rest of the methods at this point, the only one pending was `renounce_role`, after following the tutorial it shouldn't be that confusing.

Reply:
sure but there's a single element list. maybe it can be turned into a paragraph
![image](https://github.com/OpenZeppelin/cairo-contracts/assets/1703831/6075f8df-7ec6-4a7e-a23f-75ff417748fa)


> With our implementation of AccessControl, I think using the word standard without a standard will be confusing.

[there is](https://eips.ethereum.org/EIPS/eip-5982)

Reply:
> [there is](https://eips.ethereum.org/EIPS/eip-5982)

I didn't know there was a SRC for that already, nice.

Reply:
no it's not a SRC but an ERC

Reply:
```suggestion
This is the full interface of the `AccessControl` implementation:
```
in line with the Ownable suggestion